### PR TITLE
Add support for SSE job stores (fixes #1500)

### LINKED ISSUE #1500: Add support for SSE job stores
Add support for an `--sse` flag to use SSE without customer-managed keys in jobstore (see [jobStore.py:1223](https://github.com/BD2KGenomics/toil/blob/master/src/toil/jobStores/aws/jobStore.py#L1223))

### PR DESCRIPTION
**Pull request by natanlao**

### CODE REVIEW

**Comment by natanlao** [comment]
Jenkins, test this please

**Comment by natanlao** [comment]
Jenkins, test this please

File: src/toil/common.py [hunk 1]
```diff
@@ -517,9 +519,12 @@ def _addOptionFn(*name, **kwargs):
...
                 help="Enable real-time logging from workers to masters")
 
+    addOptionFn("--sse", dest="sse", default=False,
+                help="Enables server side encryption with the AWS job store. If --sseKey is not "
+                     "set, SSE (i.e. Amazon-managed encryption) will be used instead of SSE-C.")
     addOptionFn("--sseKey", dest="sseKey", default=None,
-            help="Path to file containing 32 character key to be used for server-side encryption on awsJobStore. SSE will "
-                 "not be used if this flag is not passed.")
+                help="Path to file containing 32 character key to be used for server-side encryption on "
+                     "awsJobStore. SSE will not be used if this flag is not passed.")
```
Comment (cket) [inline src/toil/common.py]: Can you clarify this? SSE-C won't be used but SSE still will be.

Comment (cket) [inline src/toil/common.py]: For backwards compatibility a user should be able to only pass this argument & we should automatically set `config.sse = True`

File: src/toil/jobStores/aws/jobStore.py [hunk 2]
```diff
@@ -849,7 +847,7 @@ def content(self, content):
 
         @classmethod
         def create(cls, ownerID):
-            return cls(str(uuid.uuid4()), ownerID, encrypted=cls.outer.sseKeyPath is not None)
+            return cls(str(uuid.uuid4()), ownerID, encrypted=cls.outer.sse)
```
Comment (natanlao) [inline src/toil/jobStores/aws/jobStore.py]: You're right - I've corrected the line in question.

**Comment by natanlao** [comment]
Comments addressed @cket

**Comment by cket** [comment]
@natanlao I'm going to add a couple tests to this PR - we shouldn't have any conflicts as long as we don't force push/ rebase!

**Comment by natanlao** [comment]
Jenkins, test this please

**Comment by natanlao** [comment]
Jenkins, test this please

**Comment by natanlao** [comment]
Jenkins, test this please

**Comment by natanlao** [comment]
Tests are failing for seemingly-unrelated reasons... `src/toil/test/src/multipartTransferTest.py::AWSMultipartCopyTest::test_multipartCopy__threadPoolSize_2` passes on my machine.

Jenkins, test this please.

**Comment by DailyDreaming** [comment]
@natanlao What's the status on this PR?

**Comment by natanlao** [comment]
@DailyDreaming If I remember correctly, it was more-or-less working around time of last commit, but fell to the backburner because of some issues with testing and because CJ left.

**Comment by natanlao** [comment]
#1500 was closed.

**Pull request closed without merging**